rest/model/conversions.go

```diff
@@ -17,56 +17,77 @@ type typeInfo struct {
 	if _, inIsPrimitive := in.(*types.Basic); inIsPrimitive {
-		return noTransforms(in.String(), outIsPtr, generatedConversions)
+		return simplePtrs(in.String(), outIsPtr, generatedConversions)
 	}
 	if intype, inIsPtr := in.(*types.Pointer); inIsPtr {
 		value, isPrimitive := intype.Elem().(*types.Basic)
 		if isPrimitive {
-			return noTransforms(value.String(), outIsPtr, generatedConversions)
+			return simplePtrs(value.String(), outIsPtr, generatedConversions)
 		}
-		return conversionFn(value, outIsPtr, generatedConversions)
+		underlyingStruct, isStruct := intype.Elem().(*types.Named)
+		if isStruct {
+			return objectTypeFnName(underlyingStruct, outIsPtr)
+		}
+		return "", "", errors.New("complex pointers not implemented yet")
 	}
 	if intype, inIsNamedType := in.(*types.Named); inIsNamedType {
 		switch intype.String() {
 		case "time.Time":
-			return noTransforms(intype.String(), outIsPtr, generatedConversions)
+			return simplePtrs(intype.String(), outIsPtr, generatedConversions)
 		default:
-			return "", errors.New("complex objects not implemented yet")
+			return objectTypeFnName(intype, outIsPtr)
 		}
 	}
 	if intype, inIsMap := in.(*types.Map); inIsMap {
 		switch intype.String() {
 		case "map[string]interface{}":
-			return noTransforms(intype.String(), outIsPtr, generatedConversions)
+			return simplePtrs(intype.String(), outIsPtr, generatedConversions)
 		default:
-			return "", errors.New("complex maps not implemented yet")
+			return "", "", errors.New("complex maps not implemented yet")
 		}
 	}
 	if intype, inIsInterface := in.(*types.Interface); inIsInterface {
 		switch intype.String() {
 		case "interface{}":
-			return noTransforms(intype.String(), outIsPtr, generatedConversions)
+			return simplePtrs(intype.String(), outIsPtr, generatedConversions)
 		default:
-			return "", errors.New("non-empty interfaces not implemented yet")
+			return "", "", errors.New("non-empty interfaces not implemented yet")
 		}
 	}
 
-	return "", errors.Errorf("converting type %s is not supported", in.String())
+	return "", "", errors.Errorf("converting type %s is not supported", in.String())
 }
 
-func noTransforms(typeName string, outIsPtr bool, generatedConversions map[string]string) (string, error) {
+func objectTypeFnName(in types.Type, outIsPtr bool) (string, string, error) {
+	fnName := ""
+	if !outIsPtr {
+		fnName = "*"
+	}
+	fnName += "API"
+	fnName += stripPackage(in.String())
+	return fnName + buildFromService, fnName + toService, nil
+}
+
+func simplePtrs(typeName string, outIsPtr bool, generatedConversions map[string]string) (string, string, error) {
 	if err := generateConversionCode(generatedConversions, typeName); err != nil {
-		return "", err
+		return "", "", err
 	}
 	typeName = cleanName(typeName)
 	if outIsPtr {
-		return fmt.Sprintf("%s%sPtr", typeName, typeName), nil
+		return fmt.Sprintf("%s%sPtr", typeName, typeName), fmt.Sprintf("%sPtr%s", typeName, typeName), nil
```

Comment:
Potentially these format strings should be constants

Reply:
I'll make them constants if I end up using them in more than 1 spot, but at the moment I don't think it's necessary